hack/generator/pkg/astmodel/resource.go

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package astmodel
+
+import (
+	"go/ast"
+	"go/token"
+)
+
+// CreateResourceDefinitions creates definitions for a resource
+func CreateResourceDefinitions(name *TypeName, specType *StructType, statusType *StructType, idFactory IdentifierFactory) (TypeDefiner, []TypeDefiner) {
+
+	var others []TypeDefiner
+
+	var specName *TypeName
+	if specType != nil {
+		specName = NewTypeName(name.PackageReference, name.Name()+"Spec")
+		specDef, specOthers := specType.CreateDefinitions(specName, idFactory)
+
+		others = append(append(others, specDef), specOthers...)
+	} else {
+		panic("spec must always be provided")
+	}
```

Comment:
Suggest inverting the condition, since `panic()` does not return ...
```suggestion
	if specType == nil {
		panic("spec must always be provided")
	}
	
	specName = NewTypeName(name.PackageReference, name.Name()+"Spec")
	specDef, specOthers := specType.CreateDefinitions(specName, idFactory)

	others = append(append(others, specDef), specOthers...)
```

Reply:
Reworked method a bit more than this, factored out common code.